.github/workflows/pypi.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will upload a Python Package using Twine when a new tag is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: detect-secrets-pypi
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+
+  tox:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python: ['3.6', '3.7', '3.8', '3.9']
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - run: python -m pip install --upgrade setuptools pip tox virtualenv
+    # Run tox only for the installed py version on the runner as outlined in the python matrix
+    # Ensures the correct py version is installed and tested as opposed to 'tox' which attempts to run for all py versions in tox.ini
+    - run: tox -e py
+    - run: tox -e mypy
+
+  deploy:
+    # lets run tests before we push anything to pypi, much like we do internally
+    needs: tox
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    # this will create a .tar.gz with all the code (i.e., an sdist)
+    - run: python setup.py sdist
+    # and finally, upload the above sdist to public PyPI
+    - uses: pypa/gh-action-pypi-publish@v1.5.0
```

Comment:
just out of curiosity, will this also upload wheels?

if you copied this from paasta, paasta doesn't distribute wheels https://pypi.org/project/paasta-tools/#files

but we do (at least, in the past we have, and IMO it's a good idea to) https://pypi.org/project/detect-secrets/#files

Reply:
Right just looked into this. There's a better way which will build the .tar and the wheels. Updating now.

Reply:
Updated to upload wheels as well.